relays/substrate/doc/design.md

```diff
@@ -0,0 +1,293 @@
+# Substrate-to-Substrate Bridge Design
```

Comment:
This seems like it will go out of date easily. I'd remove from here and keep either on a public wiki or our internal one.